### PR TITLE
admin-analytics: make "analytics" and "user management" filter state synced with URL query parameters

### PR DESCRIPTION
**Pull request by erzhtor**
Closes https://github.com/sourcegraph/sourcegraph/issues/40439.

This PR:
* Updates "analytics" and "user management" filter state to be synced with URL query parameters
* Fixes user-management reloading of legends when action end
* Adds "useURLSyncState" reusable hook + unit tests

## Test plan
- `sg start`
- Open http://localhost:3080/site-admin/users?searchText=&limit=25&lastActivePeriod=THIS_MONTH&orderBy=USERNAME&descending=true
- Check that all chart and table filters in analytics and user management pages are synced to and from URL

## App preview:

- [Web](https://sg-web-erzhtor-sync-admin-analytics.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-zvmxhicthw.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/web/src/site-admin/SiteAdminAllUsersPage/UserManagement/components/UsersList.tsx [hunk 1]
```diff
@@ -27,32 +36,47 @@ import styles from '../index.module.scss'
...
-    const [lastActivePeriod, setLastActivePeriod] = useState<SiteUsersLastActivePeriod>(SiteUsersLastActivePeriod.ALL)
-    const [limit, setLimit] = useState(LIMIT)
-    const [sortBy, setSortBy] = useState({ key: SiteUserOrderBy.EVENTS_COUNT, descending: false })
+interface UsersListProps {
+    onActionEnd?: () => void
+}
 
-    const showMore = useCallback(() => setLimit(limit => limit + LIMIT), [])
+export const UsersList: React.FunctionComponent<UsersListProps> = ({ onActionEnd }) => {
+    const [filters, setFilters] = useURLSyncedState({
```
Comment (thenamankumar) [inline client/web/src/site-admin/SiteAdminAllUsersPage/UserManagement/components/UsersList.tsx:44]: can we add/pass generic types?

File: client/wildcard/src/components/Alert/Alert.tsx [hunk 2]
```diff
@@ -26,28 +26,29 @@ const userShouldBeImmediatelyNotified = (variant?: AlertVariant): boolean =>
...
-                role={role}
-                aria-live={alertAssertiveness}
-                {...attributes}
-            >
-                {children}
-            </Component>
-        )
-    }
-) as ForwardReferenceComponent<'div', AlertProps>
+export const Alert = React.forwardRef(function Alert(
```
Comment (thenamankumar) [inline client/wildcard/src/components/Alert/Alert.tsx:29]: This is a formatting only change I guess, unnecessary for this PR.

**Review by thenamankumar — APPROVED** [review]

File: client/web/src/site-admin/SiteAdminAllUsersPage/UserManagement/components/UsersList.tsx (hunk 1 above)
Comment (erzhtor) [reply client/web/src/site-admin/SiteAdminAllUsersPage/UserManagement/components/UsersList.tsx:44]: It is already generic 👌 . In fact, this hook is reused in both analytics pages and user management.

File: client/wildcard/src/components/Alert/Alert.tsx (hunk 2 above)
Comment (erzhtor) [reply client/wildcard/src/components/Alert/Alert.tsx:29]: It is adding displayName to the component.

**Pull request merged**